apps/dashr-brain-viewer/app.R

```diff
@@ -0,0 +1,455 @@
+
+library(dashColorscales)
+library(dashR)
+library(plotly)
+library(rapportools)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(devtools)
+library(rjson)
+
+appName <- Sys.getenv("DASH_APP_NAME")
+
+appName <- Sys.getenv("DASH_APP_NAME")
+if (appName != ""){
+  pathPrefix <- sprintf("/%s/", appName)
+  
+  Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix,
+             DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+  
+  setwd(sprintf("/app/apps/%s", appName))
+}
+
+DEFAULT_COLORSCALE <- list(list(0,'#0c3383'), list(0.25,'#0a88ba'), list(0.5,'#f2d338'),
+                           list(0.75,'#f28f38'), list(1,'#d91e1e'))
+
+DEFAULT_COLORSCALE_NO_INDEX <- list()
+
+for (i in 1:length(DEFAULT_COLORSCALE)) {
+  DEFAULT_COLORSCALE_NO_INDEX[i] <- DEFAULT_COLORSCALE[[i]][2] 
+}
+
+read_mniobj <- function(file){
+  
+  triangulate_polygons <- function(list_vertex_indices){
+    j = 1
+    otherlist = list()
+    for (k in seq(1,length(list_vertex_indices),3)) {
+      otherlist[[j]] = list_vertex_indices[k:(k+2)]
+      j = j+1
+    }
+    return(otherlist)
+  }
+  
+  fp = readLines(con = file)
+  #fp=open(file,'r')
+  n_vert <- list()
+  n_poly <- list()
+  k = 0
+  j = 1
+  list_indices <- list(list())
+  fp <- as.list(fp)
+  for (i in as.numeric(labels(fp))) {
+    if (i == 1){
+      n_vert = as.numeric(as.list(strsplit(trimws(fp[i])," ")[[1]])[7])
+      vertices <- matrix(0, nrow = n_vert, ncol = 3)
+    }
+    else if(i <= n_vert) {
+      vertices[i-1,] <- unlist(lapply(strsplit(trimws(fp[i])," ")[[1]], as.numeric))
+    }    else if(i > 2*(n_vert) + 6){
+      if (lapply(fp[i], is.empty) == TRUE){
+        k = 1
+      } else if (k == 1){
+        list_indices[[j]] <- strsplit(trimws(fp[i])," ")#needs to start at one
+        j = j+1
+      }}}
+  list_indices <- lapply(as.list(unlist(list_indices)),function(x){as.integer(as.list(x))})
+  faces = as.array(triangulate_polygons(list_indices))
+  #vertices = as.matrix(vertices[[1]])
+  #vertices <- apply(X = vertices, FUN = function(x) as.list(c(x)), MARGIN = 1)
+  return(list(vertices,faces))}
+
+standard_intensity <- function(x,y,z){
+  return(z)
+}
+
+plotly_triangular_mesh <- function(vertices, faces, intensities=NULL, colorscale=NULL,
+                                   flatshading=FALSE, showscale=FALSE, reversescale=FALSE, plot_edges=FALSE){
+  
+  I = list()
+  J = list()
+  K = list()
+  for (i in 1:length(faces[[1]])) {   
+    I[i] = faces[[1]][[i]][1]
+    J[i] = faces[[1]][[i]][2]
+    K[i] = faces[[1]][[i]][3]
+  }
+  X = list()
+  Y = list()
+  Z = list()
+  X = as.list(vertices[[1]][,1])
+  Y = as.list(vertices[[1]][,2])
+  Z = as.list(vertices[[1]][,3])
+  
+  if(length(intensities) == 0){
+    intensities <- standard_intensity(x,y,z)
```

Comment:
If `standard_intensity` just returns `z`, why not get rid of that function and use `intensities <- z`?